testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/PatchRemoteHostTest.java

```diff
@@ -0,0 +1,308 @@
+ */
+public class PatchRemoteHostTest extends AbstractCliTestBase {
+    private static final int EXIT_CODE_HOST_TIMEOUT = TimeoutUtil.adjust(30);
+    protected static final PathAddress HOST_SECONDARY = PathAddress.pathAddress(HOST, "secondary");
+    protected static final PathAddress CORE_SERVICE_PATCHING = PathAddress.pathAddress(CORE_SERVICE, "patching");
+    private static final Path TARGET_DIR = Paths.get(System.getProperty("basedir", ".")).resolve("target");
+    private static DomainLifecycleUtil primaryLifecycleUtil;
+    private static DomainLifecycleUtil secondaryLifecycleUtil;
+    private static MixedDomainTestSupport support;
+    private static Path tempDir;
+    private static ProductVersion secondaryProductVersion;
+
+    @Before
+    public void init() throws Exception {
+        Assert.assertNotNull("This class is designed to be invoked from a parent test case that utilizes the @Version annotation to specify the target server version.", support);
+        tempDir = Files.createTempDirectory(TARGET_DIR, "patch-remote-host-test-");
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        if (tempDir != null && tempDir.toFile().exists()) {
+            try (Stream<Path> walk = Files.walk(tempDir)) {
+                walk.sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+            }
+        }
+    }
+
+    public static void setup(Class<?> testClass) throws IOException {
+        support = MixedDomainTestSuite.getSupport(testClass);
+
+        primaryLifecycleUtil = support.getDomainPrimaryLifecycleUtil();
+        secondaryLifecycleUtil = support.getDomainSecondaryLifecycleUtil();
+
+        WildFlyManagedConfiguration configuration = secondaryLifecycleUtil.getConfiguration();
+        secondaryProductVersion = new ProductVersion(configuration);
+    }
+
+    @Test
+    public void testPatchCommand() throws Exception {
+        try {
+            AbstractCliTestBase.initCLI(DomainTestSupport.primaryAddress);
+            // sanity check to validate we are connected to the primary DC and not to the secondary one
+            final String primaryHostName = primaryLifecycleUtil.getConfiguration().getHostName();
+            final String secondaryHostName = secondaryLifecycleUtil.getConfiguration().getHostName();
+
+            Assert.assertTrue(cli.sendLine(" :query(select=[\"host\"])", false));
+            String response = cli.readOutput();
+            Assert.assertTrue(response.contains(primaryHostName));
+            Assert.assertTrue(response.contains(secondaryHostName));
+
+            // Patch high level commands are no longer available against primary host
+            AssertionError exception = assertThrows(AssertionError.class, () -> {
+                cli.sendLine("patch info --host=" + primaryHostName);
+            });
+
+            String expectedMessage = "tool is no longer supported on this host controller";
```

Comment:
I assume this has no message id. The test should test for that instead of the non-localized text.

Reply:
@bstansberry This is a message generated by the Patch command, we are not managing CLI commands under a logger or doing any i18n with them. it doesn't mean it cannot be done via the Logger, but then some messages will be translated and some messages won't be.

Do you suggest using a logger for this case?